### PR TITLE
review refactor: Add a new way to build a query in QueryFactory

### PR DESCRIPTION
**Pull request by surli**

### CODE REVIEW

File: src/main/java/spoon/reflect/factory/QueryFactory.java [hunk 1]
```diff
@@ -48,4 +48,12 @@ public CtQuery createQuery() {
 	public CtQuery createQuery(Object input) {
 		return new CtQueryImpl(input);
 	}
+
+	/**
+	 * Creates a bound query with an optional number of inputs
```
Comment (monperrus) [inline src/main/java/spoon/reflect/factory/QueryFactory.java]: what is the meaning of inputs?

not fan of pointing to an implementation in an interface.

**Comment by monperrus** [comment]
interesting. it seems we need a test case.

**Comment by surli** [comment]
@monperrus @pvojtechovsky if you're ok with this one, it's ready for merging.

**Comment by monperrus** [comment]
ok for me. @pvojtechovsky do you merge?

**Pull request merged**